### ISSUE TITLE
msmtpq fails with `can't find the msmtp executable [ msmtp ]`

### ISSUE THREAD

**Issue by cresh** [NixOS/nixpkgs#188380]
### Describe the bug
running `msmtpq` fails with `can't find the msmtp executable [ msmtp ]`

### Steps To Reproduce

1. Run `msmtpq --q-mgmt`

The command fails with `can't find the msmtp executable [ msmtp ]`.

(Actually, it fails with an error that is caused by https://github.com/marlam/msmtp-mirror/issues/93, but it boils down to `executable not found`.)

### Expected behavior

The above command should output 

```
  no mail in queue

```

if no mail is in queue.

### Notify maintainers

@peterhoeg 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.5.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.10.3`
 - channels(root): `"nixpkgs"`
 - channels(crestani): `"nixpkgs"`
 - nixpkgs: `/Users/crestani/nixpkgs`
```

**Comment by cresh** [comment]
I'm not familiar with `resholve`, but I gave it a try and came up with this PR that works for me:
https://github.com/NixOS/nixpkgs/pull/188379

**Comment by peterhoeg** [comment]
Fixed in f7d1697db266f98093efd2a5e5bc98a29fa5affb
